openmetadata-ui/src/main/resources/ui/src/pages/RolesPage/AddRuleModal.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ *  Copyright 2021 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import classNames from 'classnames';
+import { isUndefined } from 'lodash';
+import { FormErrorData } from 'Models';
+import React, { FC, useState } from 'react';
+import { Button } from '../../components/buttons/Button/Button';
+import { RuleAccess } from '../../enums/rule.enum';
+import {
+  Operation,
+  Rule,
+} from '../../generated/entity/policies/accessControl/rule';
+import { errorMsg } from '../../utils/CommonUtils';
+
+interface AddRuleProps {
+  header: string;
+  errorData?: FormErrorData;
+  initialData: Rule;
+  onCancel: () => void;
+  onSave: (data: Rule) => void;
+}
+
+const AddRuleModal: FC<AddRuleProps> = ({
+  onCancel,
+  header,
+  initialData,
+  errorData,
+  onSave,
+}: AddRuleProps) => {
+  const [data, setData] = useState<Rule>(initialData);
+  const [access, setAccess] = useState<RuleAccess>(
+    initialData.allow ? RuleAccess.ALLOW : RuleAccess.DENY
+  );
+  const [isEnabled, setIsEnabled] = useState<boolean>(
+    Boolean(initialData.enabled)
+  );
+  const onChangeHadler = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    e.persist();
+    setData((prevState) => {
+      return {
+        ...prevState,
+        [e.target.name]: e.target.value,
+      };
+    });
+  };
+
+  const onSubmitHandler = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    onSave({
+      ...data,
+      allow: access === RuleAccess.DENY,
```

Comment:
allow should be true if `access === RuleAccess.ALLOW` ?

Reply:
Yeah, changes are in my local.